codegens/nodejs-axios/lib/axios.js

```diff
@@ -0,0 +1,225 @@
+ *
+ * @param {Object} request - Postman SDK request object
+ * @param {String} indentString - indentation required for code snippet
+ * @param {Object} options
+ * @returns {String} - nodejs(axios) code snippet for given request object
+ */
+function makeSnippet (request, indentString, options) {
+
+  var snippet,
+    configArray = [],
+    dataSnippet = '',
+    body,
+    headers,
+    indentType = options.indentType === 'Tab' ? '\t' : ' ',
+    indent = indentType.repeat(options.indentCount),
+    varDeclare = options.ES6_enabled ? 'const' : 'var';
+
+  snippet = varDeclare + ' axios = require(\'axios\');\n';
+  if (request.body && !request.headers.has('Content-Type')) {
+    if (request.body.mode === 'file') {
+      request.addHeader({
+        key: 'Content-Type',
+        value: 'text/plain'
+      });
+    }
+    else if (request.body.mode === 'graphql') {
+      request.addHeader({
+        key: 'Content-Type',
+        value: 'application/json'
+      });
+    }
+  }
+
+  // The following code handles multiple files in the same formdata param.
+  // It removes the form data params where the src property is an array of filepath strings
+  // Splits that array into different form data params with src set as a single filepath string
+  if (request.body && request.body.mode === 'formdata') {
+    let formdata = request.body.formdata,
+      formdataArray = [];
+    formdata.members.forEach((param) => {
+      let key = param.key,
+        type = param.type,
+        disabled = param.disabled,
+        contentType = param.contentType;
+      // check if type is file or text
+      if (type === 'file') {
+        // if src is not of type string we check for array(multiple files)
+        if (typeof param.src !== 'string') {
+          // if src is an array(not empty), iterate over it and add files as separate form fields
+          if (Array.isArray(param.src) && param.src.length) {
+            param.src.forEach((filePath) => {
+              addFormParam(formdataArray, key, param.type, filePath, disabled, contentType);
+            });
+          }
+          // if src is not an array or string, or is an empty array, add a placeholder for file path(no files case)
+          else {
+            addFormParam(formdataArray, key, param.type, '/path/to/file', disabled, contentType);
+          }
+        }
+        // if src is string, directly add the param with src as filepath
+        else {
+          addFormParam(formdataArray, key, param.type, param.src, disabled, contentType);
+        }
+      }
+      // if type is text, directly add it to formdata array
+      else {
+        addFormParam(formdataArray, key, param.type, param.value, disabled, contentType);
+      }
+    });
+
+    request.body.update({
+      mode: 'formdata',
+      formdata: formdataArray
+    });
+  }
+  
+  body = request.body && request.body.toJSON();
+    
+  dataSnippet = parseRequest.parseBody(body, 
```

Comment:
only add the body if it is non empty object, and the mode that it states has non empty data

Reply:
done